### DISCUSSION TITLE
why is Connection::execute hidden in doc?

### DISCUSSION THREAD

**Discussion by keepsimple1** [diesel-rs/diesel#3063] [Q&A] [answered]
https://github.com/diesel-rs/diesel/blob/4cae65cb02cd1c50cf3173505df497fc32d64d56/diesel/src/connection/mod.rs#L214-L215

I'm curious why this method is hidden in doc?

**Comment by weiznich** [comment, marked as answer]
`#[doc(hidden)]` methods are generally not considered to be part of the public API of an crate. That means they are free to change at any point in time without warning or issuing a major version bump. 
As for `Connection::execute`: This method is just an implementation detail, which is used internally by diesel. As this method is part of an public trait we just hide this (and other) methods so that they are not part of the public API.
